spec/thread_safety_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe "thread safety" do # rubocop:disable RSpec/DescribeClass
+  context "when two threads accessing unmemoized zero-args method" do
+    # Tip of the hat to @jeremyevans,
+    # see https://github.com/panorama-ed/memo_wise/pull/224
+    let(:thread_return_values) do
+      thread_return_values = [nil, nil]
+
+      1000.times do
+        instance = class_with_memo.new
+
+        threads = Array.new(2).map do
```

Comment:
I think the `map` is superfluous with `Array.new(2)` as well. `Array#new` can optionally take a block, so we could do:

```ruby
Array.new(2) do
   ...
end
```